src/Service.Tests/Configuration/TelemetryTests.cs

```diff
@@ -0,0 +1,234 @@
+        //     item is RequestTelemetry
+        //     && ((RequestTelemetry)item).Name.Equals("POST /graphql")
+        //     && ((RequestTelemetry)item).ResponseCode.Equals("200")
+        //     && ((RequestTelemetry)item).Url.PathAndQuery.Equals("/graphql")));
+
+        // Assert.IsTrue(telemetryItems.Any(item =>
+        //     item is RequestTelemetry
+        //     && ((RequestTelemetry)item).Name.Equals("POST Rest/Insert [route]")
+        //     && ((RequestTelemetry)item).ResponseCode.Equals("403")
+        //     && ((RequestTelemetry)item).Url.PathAndQuery.Equals("/api/Publisher/id/1?name=Test")));
+
+        // Assert on the Exceptions telemetry items.
+        Assert.AreEqual(1, telemetryItems.Count(item => item is ExceptionTelemetry));
+        Assert.IsTrue(telemetryItems.Any(item =>
+            item is ExceptionTelemetry
+            && ((ExceptionTelemetry)item).Message.Equals("Authorization Failure: Access Not Allowed.")));
+    }
+
+    /// <summary>
+    /// Validates that no telemetry data is sent to CustomTelemetryChannel when 
+    /// Appsights is disabled OR when no valid connectionstring is provided.
+    /// </summary>
+    /// <param name="isTelemetryEnabled">Whether telemetry is enabled or not.</param>
+    /// <param name="telemetryConnectionString">Telemetry connection string.</param>
+    [DataTestMethod]
+    [DataRow(false, "", DisplayName = "Configuration without a connection string and with Application Insights disabled.")]
+    [DataRow(true, "", DisplayName = "Configuration without a connection string, but with Application Insights enabled.")]
+    [DataRow(false, TEST_APP_INSIGHTS_CONN_STRING, DisplayName = "Configuration with a connection string, but with Application Insights disabled.")]
+    public async Task TestNoTelemetryItemsSentWhenDisabled_NonHostedScenario(bool isTelemetryEnabled, string telemetryConnectionString)
+    {
+        SetUpTelemetryInConfig(CONFIG_WITHOUT_TELEMETRY, isTelemetryEnabled, telemetryConnectionString);
+
+        string[] args = new[]
+        {
+           $"--ConfigFileName={CONFIG_WITHOUT_TELEMETRY}"
+       };
+
+        ITelemetryChannel telemetryChannel = new CustomTelemetryChannel();
+        Startup.CustomTelemetryChannel = telemetryChannel;
+
+        using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+        {
+            await TestRestAndGraphQLRequestsOnServerInNonHostedScenario(server);
+        }
+
+        List<ITelemetry> telemetryItems = ((CustomTelemetryChannel)telemetryChannel).GetTelemetryItems();
+
+        // Assert that we are not sending any Traces/Requests/Exceptions to Telemetry
+        Assert.IsTrue(telemetryItems.IsNullOrEmpty());
+    }
+
+    /// <summary>
+    /// This method tests the ability of the server to handle GraphQL and REST requests,
+    /// and verifies that the server returns the expected response status codes.
+    /// Makes a valid GraphQL Request and one Invalid REST Request.
+    /// </summary>
```

Comment:
Can you elaborate on how this relates to the Telemetry tests? ie why successful execution validates telemetry config

Reply:
this method is just used as helper for other test methods to execute REST and GRaphQL requests

Reply:
In that case, the description and method name should be updated to reflect that. Right now the description and name read as if this were its test.

I don't think the asserts need to be kept either. Essentially the point of this method is to issue requests which trigger the logging system to emit logs.

Reply:
Gotcha